### PR TITLE
crosscluster/logical: only return jobID if planner txn succeeded

### LINKED ISSUE #135176: ccl/crosscluster/logical: TestFourWayReplication failed
ccl/crosscluster/logical.TestFourWayReplication [failed](https://mesolite.cluster.engflow.com/invocations/default/dfb3c626-5d9d-4991-bdf3-0cd82b3dc36a?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [028f96fbccf525df470e41ef58797935393e4c5d](https://github.com/cockroachdb/cockroach/commits/028f96fbccf525df470e41ef58797935393e4c5d):

```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFourWayReplication4205326884
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1020815129178767361 to reach replicated time of 1731598457.448267844,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020815129825280001 to reach replicated time of 1731598457.448267844,0
    jobs_verification.go:218: error scanning '&{<nil> 0xc00483b7a0}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestFourWayReplication4205326884
--- FAIL: TestFourWayReplication (4.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-44393

### LINKED ISSUE #135490: ccl/crosscluster/logical: TestMultipleSourcesIntoSingleDest failed
ccl/crosscluster/logical.TestMultipleSourcesIntoSingleDest [failed](https://tanzanite.cluster.engflow.com/invocations/default/ccd997f6-b273-4546-9aae-1261f5d630aa?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [e83bc46aa42f2476b4b11b9703b8038c660dc980](https://github.com/cockroachdb/cockroach/commits/e83bc46aa42f2476b4b11b9703b8038c660dc980):

```
=== RUN   TestMultipleSourcesIntoSingleDest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestMultipleSourcesIntoSingleDest3604595356
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1021337637439668225 to reach replicated time of 1731757912.711583888,0
    jobs_verification.go:218: error scanning '&{<nil> 0xc007bda870}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestMultipleSourcesIntoSingleDest3604595356
--- FAIL: TestMultipleSourcesIntoSingleDest (14.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultipleSourcesIntoSingleDest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-44591

### PR DESCRIPTION
**Pull request by msbutler**
PR #134997 introduced a bug where CREATE LOGICAL REPLICATION could return early with a bogus job id if the txn which writes the job record retried. This patch prevents this.

Note: I plan to open a follow up PR that moves the ldr planning txn into its own helper func.

Fixes #135176
Fixes #135490

Release note: none

### CODE REVIEW

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go [hunk 1]
```diff
@@ -280,6 +280,14 @@ func createLogicalReplicationStreamPlanHook(
 		if cr, ok := options.GetDefaultFunction(); ok {
 			defaultConflictResolution = *cr
 		}
+
+		var jobID jobspb.JobID
```
Comment (jeffswenson) [inline pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:284]: nit: I find deferred control flow hard to follow when reading code. This can be written more clearly using standard error handling.

var jobID jobspb.JobID
err := p.ExecCfg(... {

}
if err != nil {
   return err
}
resultCh <- tree.Datums{tree.NewDInt(tree.DInt(jobID))}
return nil

Comment (msbutler) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:284]: Done.

**Review by kev-cao — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=kev-cao
